mycroft/messagebus/client/client.py

```diff
@@ -0,0 +1,216 @@
+# Copyright 2017 Mycroft AI Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import json
+import time
+import traceback
+from threading import Event
+
+from websocket import (
+    WebSocketApp,
+    WebSocketConnectionClosedException,
+    WebSocketException
+)
+
+from mycroft.messagebus.load_config import load_message_bus_config
+from mycroft.messagebus.message import Message
+from mycroft.util import create_echo_function
+from mycroft.util.log import LOG
+from .threaded_event_emitter import ThreadedEventEmitter
+
+
+class MessageBusClient(object):
+    def __init__(self, host=None, port=None, route=None, ssl=None):
+        config_overrides = dict(host=host, port=port, route=route, ssl=ssl)
+        self.config = load_message_bus_config(**config_overrides)
+        self.emitter = ThreadedEventEmitter()
+        self.client = self.create_client()
+        self.retry = 5
+        self.connected_event = Event()
+        self.started_running = False
+
+    def create_client(self):
+        url = '{scheme}://{host}:{port}{route}'.format(
+            scheme='wss' if self.config.ssl else 'ws',
+            host=self.config.host,
+            port=str(self.config.port),
+            route=self.config.route
+        )
+        return WebSocketApp(
+            url,
+            on_open=self.on_open,
+            on_close=self.on_close,
+            on_error=self.on_error,
+            on_message=self.on_message
+        )
+
+    def on_open(self):
+        LOG.info("Connected")
+        self.connected_event.set()
+        self.emitter.emit("open")
+        # Restore reconnect timer to 5 seconds on sucessful connect
+        self.retry = 5
+
+    def on_close(self):
+        self.emitter.emit("close")
+
+    def on_error(self, error):
+        """ On error start trying to reconnect to the websocket. """
+        if isinstance(error, WebSocketConnectionClosedException):
+            LOG.warning('Could not send message because connection has closed')
+        else:
+            LOG.exception('=== ' + repr(error) + ' ===')
+
+        try:
+            self.emitter.emit('error', error)
+            if self.client.keep_running:
+                self.client.close()
+        except Exception as e:
+            LOG.error('Exception closing websocket: ' + repr(e))
+
+        LOG.warning("WS Client will reconnect in %d seconds." % self.retry)
```

Comment:
Since this is now MessageBusclient should we change this warning to `Messagbus client will reconnect ...`